k8s/pkg/apis/longhorn/v1beta2/recurringjob.go

```diff
@@ -17,14 +17,21 @@ type VolumeRecurringJob struct {
 	IsGroup bool   `json:"isGroup"`
 }
 
+// RecordVolumeRecurringJob defines the Longhorn recurring job information stroed into the backup volume configuration.
+type RecordVolumeRecurringJob struct {
+	JobSpec   RecurringJobSpec `json:"jobSpec"`
+	FromGroup []string         `json:"fromGroup,omitempty"`
+	FromJob   bool             `json:"fromJob"`
+}
+
 // RecurringJobSpec defines the desired state of the Longhorn recurring job
 type RecurringJobSpec struct {
 	// The recurring job name.
 	// +optional
 	Name string `json:"name"`
 	// The recurring job group.
 	// +optional
-	Groups []string `json:"groups,omitempty"`
```

Comment:
Modified, not remove `omitempty`, set `job.JobSpec.Groups` to be `[]string{}` if `JobSpec.Groups` from backup target is `nil` 

It would get `[]string = nil` when decoding recurring job information from the backup target would be different to `[]string{}` of recurring jobs CR in the cluster judged by `reflect.DeepEqual` And then it would create a new recurring job every time because the field `groups` is not the same even they both are empty. 
It is equal when I encode the same structure and decode it locally.